docs/dsms_sdk/dsms_config_schema.md

```diff
@@ -9,13 +9,15 @@ This section describes the configuration properties for the DSMS Python SDK.
 |-----------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------------------|------------------------------|-------------------------|-------------------|-------------------------------|
 | Host URL              | URL of the DSMS instance to connect.                                                                                                                                                                                                                                                                | `AnyUrl`             | Not Applicable               | `host_url`              | Required          | `DSMS_HOST_URL`               |
-| Request timeout       | Timeout in seconds until the request to the DSMS is timed out.                                                                                                                                                                                                                                      | `int`                | `120`                        | `request_timeout`       | Optional          | `DSMS_REQUEST_TIMEOUT`        |
-| SSL verify            | Whether the SSL of the DSMS shall be verified during connection.                                                                                                                                                                                                                                    | `bool`               | `True`                       | `ssl_verify`            | Optional          | `DSMS_SSL_VERIFY`             |
 | Username              | User name for connecting to the DSMS instance                                                                                                                                                                                                                                                       | `Optional[SecretStr]`| `None`                       | `username`              | Optional          | `DSMS_USERNAME`               |
+| Client ID | If a service account is used to authenticate, this will proviode the Client ID in Keycloak |  `Optional[SecretStr]`| `None`                       | `client_id`              | Optional          | `DSMS_CLIENT_ID`, `KEYCLOAK_DSMS_CLIENT_ID` or `KEYCLOAK_CLIENT_ID`               |
+| Client Secret | If a service account is used to authenticate, this will proviode the Client Secret in Keycloak |  `Optional[SecretStr]`| `None`                       | `client_secret`              | Optional          | `DSMS_CLIENT_SECRET`, `KEYCLOAK_DSMS_CLIENT_SECRET` or `KEYCLOAK_CLIENT_SECRET`               |
```

Comment:
Can you explain why did you add these variables? To avoid using the DSMS admin user? Does it mean we need to share this client secret to all users?

Reply:
The DSMS Admin user is a hack which we identified to remove. It can also cause problems with the 2FA during internal processes in the backend. One instead can simply authenticate through the client secret with the serivce account, which is automatically enabled in Keycloak for each client. 

I do not understand where you got the impression from that we are sharing the client secret with all users. The user has the option to authenticate with:

- username/password
- JWT
- client id/client secret

The latter one does not get exposed when using the first two one. The only place where we are using the third option is in the backend, where the client secret should also not be leaked from.